dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +249,198 @@ sales_val AS (
         sf.run_id AS sv_run_id,
         sf.version AS sv_version
     FROM
-        {{ source('sale', 'flag') }}
-            AS sf
+        {{ source('sale', 'flag') }} AS sf
     INNER JOIN max_version_flag AS mv
         ON sf.meta_sale_document_num = mv.meta_sale_document_num
         AND sf.version = mv.max_version
+),
+
+-- CTE to coalesce iasworld and mydec values prior to
+-- constructing filters that depend on coalesced fields
+combined_sales AS (
+    SELECT
+        COALESCE(uq_sales.pin, md_sales.pin) AS pin_coalesced,
+        -- For many of the fields we used  simple coalesce statement,
+        -- but some data is a bit more complicated. Prior to 2021,
+        -- mydec sales and iasworld sales used different sale dates.
+        -- We preference the mydec sale as they are believed to be more
+        -- accurate. As of 2021, iasworld utilizes mydec sales, which means
+        -- we can prioritize iasworld data instead of mydec data.
+        CASE
+            WHEN uq_sales.year < '2021'
+                THEN COALESCE(md_sales.year, uq_sales.year)
+            ELSE COALESCE(uq_sales.year, md_sales.year)
+        END AS year_coalesced,
+        COALESCE(uq_sales.township_code, tc.township_code)
+            AS township_code_coalesced, --noqa
+        COALESCE(uq_sales.nbhd, tc.nbhd) AS nbhd_coalesced,
+        COALESCE(uq_sales.class, tc.class) AS class_coalesced,
+        CASE --noqa
+            WHEN
+                uq_sales.year < '2021'
+                THEN COALESCE(md_sales.sale_date, uq_sales.sale_date)
+            ELSE COALESCE(uq_sales.sale_date, md_sales.sale_date)
+        END AS sale_date_coalesced,
+        CASE
+        -- If uq_sales.doc_no is not NULL, apply the COALESCE logic
+            WHEN uq_sales.doc_no IS NOT NULL
+                THEN COALESCE(
+                    md_sales.sale_date IS NOT NULL
+                    OR YEAR(uq_sales.sale_date) >= 2021,
+                    FALSE
+                )
+            ELSE
+                TRUE
+        END AS is_mydec_date,
```

Comment:
I'm not sure this logic is correct. In the past, `is_mydec_date` indicated at least 1 of 2 possible things:
- there was mydec data available for this sale, so we preferred the mydec date
- it was 2021 or later, when iasworld starts using mydec dates

if i grab the new view only for source = 'iasworld' and tally `is_mydec_date`, it currently doesn't match the old view.

I think you should only need the inner `coalesce` function and not the `case when`?

Reply:
Fixed this. I had accidentally removed a multi-parcel filter a few commits back